promql/testdata/subquery.test

```diff
@@ -0,0 +1,182 @@
+## The following tests are from selectors.test
```

Comment:
It'd be best to focus on aspects of the implementation, rather than things that test the internals of other functionality.

For example, how does offset work with this?

Reply:
I had not added support for offset for subquery. Should I add it?